packages/headless/src/commerce.index.ts

```diff
@@ -216,6 +216,8 @@ export type {
   RecentQueriesState,
 } from './controllers/recent-queries-list/headless-recent-queries-list.ts';
 
+export type {RegisterRecentQueriesCreatorPayload} from './features/recent-queries/recent-queries-actions';
```

Comment:
Instead of exporting this type directly in `commerce.index.ts`, we could export it in `recent-queries-actions-loader.ts`. Then since we're already doing

```
export * from './features/commerce/recent-queries/recent-queries-actions-loader';
```

on line 61, we'd be exporting the type at the same time!